.github/workflows/crate-check.yml

```diff
@@ -22,7 +22,7 @@ jobs:
       - name: Get commit range (push)
         if: ${{ github.event_name == 'push' }}
         run: |
-          echo "COMMIT_RANGE=$GITHUB_SHA" >> $GITHUB_ENV
+          echo "COMMIT_RANGE=HEAD~1..$GITHUB_SHA" >> $GITHUB_ENV
```

Comment:
The `push` event could have multiple commits though. The docs for the [push webhook](https://docs.github.com/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push) indicate there's a "before" parameter that could potentially be used here instead of hard coding `HEAD~1`

Reply:
yeah, thank you for pointing this out! modified here [0febc8c](https://github.com/solana-labs/solana/pull/29755/commits/0febc8cd9cb8412a6016e51af44c8f77f55c2358)